src/main/java/deku/task/Task.java

```diff
@@ -0,0 +1,116 @@
+package deku.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+
+import deku.DekuExceptions;
+import deku.InputParser;
+
+/**
+ * Main parent Task class
+ */
+public class Task {
+    private final List<String> taskArray;
+    private final String task;
+    private final String icon;
+    private Boolean completionStatus;
+    private String completionIcon;
+    private final InputParser parser = new InputParser();
+
+    Task(List<String> task, String taskName, String icon) throws DekuExceptions {
+        if (task.size() == 0) {
+            throw new DekuExceptions("The description of a " + taskName + " cannot be empty.");

```

Comment:
>In general, try to avoid any magic literals.

Consider making a constant variable for this instead or create an `EmptyDescriptionException` class that inherits `DekuExceptions` for this.